src/storage/v2/storage.cpp

```diff
@@ -1144,6 +1144,13 @@ void Storage::Accessor::FinalizeTransaction() {
   }
 }
 
+std::optional<uint64_t> Storage::Accessor::GetTransactionId() const {
+  if (is_transaction_active_) {
+    return transaction_.transaction_id;
+  }
+  return {};
```

Comment:
I think this is not responsibility of this function to return id only if transaction is active. In my opinion we should return `id` but other side should check if transaction is active, and other stuff which we need to do. This seems like there is some hidden logic there.

Reply:
Yes but if there is no active transaction, there is nothing to return just garbage and hence the program can crash with exception

Reply:
Okay, why do we delete the transaction id? I didn't know that happens.

Reply:
If in the meantime, transaction ended and new wasn't started.